### PR TITLE
WIP add interface converter

### PR DESCRIPTION
**Pull request by rf**
Adds interface converter, which can handle a logtron or larch.

Needs to accept `DebugLogtron`s which don't have the same interface as `Logtron` :(

r @Raynos

### CODE REVIEW

File: interface-converter.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+var LogtronBackend = require('./logtron-backend');
+var Larch = require('./larch');
+
+module.exports = createLarchWithLogger;
+
+function createLarchWithLogger(config) {
+    assert(config.logger, 'createLarchWithLogger requires a logger');
+
+    var backendLogger = config.logger;
+    delete config.logger;
```
Comment (Raynos) [inline interface-converter.js]: no >:(. no `delete`. not allowed.

File: interface-converter.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    if (isLogtronLogger(backendLogger)) {
+        var logtronBackend = new LogtronBackend(backendLogger);
+        config.backends = [logtronBackend];
+        return new Larch(config);
```
Comment (Raynos) [inline interface-converter.js]: just create a new config object...

**Comment by Raynos** [comment]
Seems legit. Please leave comment about `delete`

**Comment by jcorbin** [comment]
This happens once at process start time right? So the delete is :ok: with me.

**Comment by walterddr** [comment]
Hi Rong,

I would love to have you join my dinner. There are still a couple more
spots as of now. I will send you more details as the date gets closer.
Looking forward to meeting you soon.

Go Bruins!

Victoria

**Comment by walterddr** [comment]
hi rong

**Comment by ahhh** [comment]
Sorry for any interruption / wayward comments in this thread. Security Response had detected an issue and then performed some tests in the comment field here. The issue has since been remediated and there should be no further issues or cause for concern. Thank you for understanding.

**Pull request merged**